src/doom/am_map.c

```diff
@@ -674,6 +680,18 @@ void AM_LevelInit(boolean reinit)
                 color_shades[color * NUMSHADES + shade] = colormaps[shade_index[shade]];
             }
         }
+        if (original_playpal)
```

Comment:
Please put away *all* of these conditionals and initialize those two values independent of the palette being from the IWAD or not. In case of the IWAD palette, the exact same color as before will be selected. In all other cases, it will return the closest one. No need for any more logic.

Reply:
You might want to initialize the statics to `-1` (impossible palette index) and make sure the code is only run once.

Reply:
> You might want to initialize the statics to `-1` (impossible palette index) and make sure the code is only run once.

The code runs on `AM_LevelInit` inside the conditional
`
if (!precalc_once)
{
    precalc_once = 1;
}
`
which means it runs only once (`precalc_once = 1` prevents fulfilling of the if condition at the next run).